### ISSUE TITLE
Path separators in show and episode titles should be substituted.

### ISSUE THREAD

**Issue by shinji257** [anidl/multi-downloader-nx#334]
### Program version

3.0.7

### Operating System

Linux

### Type

Cli

### Service

Crunchyroll

### Command used

./aniDL --service crunchy --s GR2484X76 --all

### Show ID

GR2484X76

### Episode

12

### Console Output

N/A

### Additional Information

```Text
Path separators (/ and \) should be replaced with an underscore like other potentially invalid characters.  This avoids a scenario (at least on linux) where it creates additional sub folders because the various supporting programs create additional sub folders during downloading.  Luckily it still downloads and processes properly albeit with a bit of un-necessary path splitting.  I can manually fix it but I'd rather not have to do that if I don't need to.
```

**Comment by shinji257** [comment]
Add quotes (both single and double) to this as well.  I just hit an episode where a single quote in the filename broke processing later.

```
[INFO] Requesting: [GRZJ07X96] 8 Man After (Sub) - 4 - Sachiko`s Decision
[INFO] Selecting stream with English hardsubs
[INFO] Full stream found! (✓1: adaptive_hls/jpn/en )
[INFO] Full stream found! ( 2: download_hls/jpn/en )
[INFO] Full stream found! ( 3: vo_adaptive_hls/jpn/en )
[INFO] Downloading video...
[INFO] Playlists URL: https://pl.crunchyroll.com/evs3/43def502cd568274595e8a0544a06d74/assets/2600144e6c2b30eb74619ae08ffdbe83_,3668692.mp4,3668690.mp4,3668691.mp4,3668689.mp4,.urlset/master.m3u8?Expires=1656722886&Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiaHR0cHM6Ly9wbC5jcnVuY2h5cm9sbC5jb20vZXZzMy80M2RlZjUwMmNkNTY4Mjc0NTk1ZThhMDU0NGEwNmQ3NC9hc3NldHMvMjYwMDE0NGU2YzJiMzBlYjc0NjE5YWUwOGZmZGJlODNfLDM2Njg2OTIubXA0LDM2Njg2OTAubXA0LDM2Njg2OTEubXA0LDM2Njg2ODkubXA0LC51cmxzZXQvbWFzdGVyLm0zdTgiLCJDb25kaXRpb24iOnsiRGF0ZUxlc3NUaGFuIjp7IkFXUzpFcG9jaFRpbWUiOjE2NTY3MjI4ODZ9fX1dfQ__&Signature=CgWEDNXN7LLNMfIxxdrjQFZEgg3EcOT0QfWJc~8I7rFmUW1glgHmEj0bL5nkAPyCm5fYz~B5fU3H6b~Ib1~4DRaCRhZ-~fe~-P-A339wVXheLvoSZFvoxhRO72gXOsGeKFtWIXq41AeCSoUOgicSfJOQDHy8WX506mSUoSMH86xK3UEQAGTvgiHOVqDFMWJJYJU5IA6gdXLlZcV5QrcgctzEdg0d5Qvp4MHgwF7iJwx~dcQ7qemeThXhJ0hN~ZmZda5J14nTHxxgT0q8A~9hb57fwW2A~FdhQFzDDELp~d97lDUgDixD7yh7-F~NsenzxaBR07Ur9IUnRKGsqq0-QQ__&Key-Pair-Id=APKAJMWSQ5S7ZB3MF5VA (adaptive_hls/jpn/en)
[INFO] Servers available:
        pl.crunchyroll.com (ll-prod)
        pl.crunchyroll.com (akamai-prod)
[INFO] Available qualities:
        [1] 128x96 (191KiB/s)
        [2] 320x240 (549KiB/s)
        [3] 464x360 (1070KiB/s)
        [4] 624x480 (2043KiB/s)
[INFO] Selected quality: 624x480 @ pl.crunchyroll.com (ll-prod)
[INFO] Stream URL: https://pl.crunchyroll.com/evs3/43def502cd568274595e8a0544a06d74/assets/2600144e6c2b30eb74619ae08ffdbe83_3668692.mp4/index-v1-a1.m3u8?Expires=1656722887&Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiaHR0cHM6Ly9wbC5jcnVuY2h5cm9sbC5jb20vZXZzMy80M2RlZjUwMmNkNTY4Mjc0NTk1ZThhMDU0NGEwNmQ3NC9hc3NldHMvMjYwMDE0NGU2YzJiMzBlYjc0NjE5YWUwOGZmZGJlODNfKi5tM3U4PyoiLCJDb25kaXRpb24iOnsiRGF0ZUxlc3NUaGFuIjp7IkFXUzpFcG9jaFRpbWUiOjE2NTY3MjI4ODd9fX1dfQ__&Signature=GuRSwl00OQN~ZeGnVBI-kUkIDo5akJwd-ext7GVqM26kHrEI5putBKMQGh5bvnxxqfH2yG3HoQ9FXEtG4DFFwr9n8j6umNV2o0ZLARhJ8r6y8mRU7WmK4t-WP58pJyrVXZXLw41L7dVP4NV5~-I7IhAfrcsnC1vI4e9Tya1J3YErmW3EusmTTcoDPR-LFTnnkr0cyKAdT58HZF06tT0vIFXEKd-2y4EmC5FSgcPi~YI9RljGFhvoTKFee7kK-989j1dCr3lOC8ws5DY9onMh6w3QI9-UVH221uTt3qtUAqxwKEiXTjDIWVnZD59trtv9LNUI48--pj0M~4y8VzTfsg__&Key-Pair-Id=APKAJMWSQ5S7ZB3MF5VA&cdn=ll-prod
[INFO] Output filename: 8 Man After (Sub)/S01E04 - Sachiko`s Decision [480p].Japanese
[INFO] Total parts in stream: 422 (43*10)
[INFO] Saving stream to «/home/shinji/multi-downloader-nx/videos/8 Man After (Sub)/S01E04 - Sachiko`s Decision [480p].Japanese.ts»...
[INFO] 10 of 422 parts downloaded [2%] (1m11s | 6.1Mb/s)
[INFO] 20 of 422 parts downloaded [5%] (48s | 7.9Mb/s)
[INFO] 30 of 422 parts downloaded [7%] (43s | 9.3Mb/s)
[INFO] 40 of 422 parts downloaded [9%] (40s | 10Mb/s)
[INFO] 50 of 422 parts downloaded [12%] (35s | 11Mb/s)
[INFO] 60 of 422 parts downloaded [14%] (32s | 12Mb/s)
[INFO] 70 of 422 parts downloaded [17%] (29s | 13Mb/s)
[INFO] 80 of 422 parts downloaded [19%] (27s | 14Mb/s)
[INFO] 90 of 422 parts downloaded [21%] (26s | 14Mb/s)
[INFO] 100 of 422 parts downloaded [24%] (24s | 14Mb/s)
[INFO] 110 of 422 parts downloaded [26%] (23s | 14Mb/s)
[INFO] 120 of 422 parts downloaded [28%] (23s | 14Mb/s)
[INFO] 130 of 422 parts downloaded [31%] (22s | 14Mb/s)
[INFO] 140 of 422 parts downloaded [33%] (22s | 13Mb/s)
[INFO] 150 of 422 parts downloaded [36%] (21s | 13Mb/s)
[INFO] 160 of 422 parts downloaded [38%] (20s | 13Mb/s)
[INFO] 170 of 422 parts downloaded [40%] (19s | 13Mb/s)
[INFO] 180 of 422 parts downloaded [43%] (19s | 13Mb/s)
[INFO] 190 of 422 parts downloaded [45%] (18s | 13Mb/s)
[INFO] 200 of 422 parts downloaded [47%] (17s | 13Mb/s)
[INFO] 210 of 422 parts downloaded [50%] (16s | 13Mb/s)
[INFO] 220 of 422 parts downloaded [52%] (16s | 13Mb/s)
[INFO] 230 of 422 parts downloaded [55%] (15s | 13Mb/s)
[INFO] 240 of 422 parts downloaded [57%] (14s | 14Mb/s)
[INFO] 250 of 422 parts downloaded [59%] (13s | 14Mb/s)
[INFO] 260 of 422 parts downloaded [62%] (12s | 14Mb/s)
[INFO] 270 of 422 parts downloaded [64%] (12s | 14Mb/s)
[INFO] 280 of 422 parts downloaded [66%] (11s | 14Mb/s)
[INFO] 290 of 422 parts downloaded [69%] (10s | 14Mb/s)
[INFO] 300 of 422 parts downloaded [71%] (9s | 14Mb/s)
[INFO] 310 of 422 parts downloaded [73%] (9s | 14Mb/s)
[INFO] 320 of 422 parts downloaded [76%] (8s | 14Mb/s)
[INFO] 330 of 422 parts downloaded [78%] (7s | 14Mb/s)
[INFO] 340 of 422 parts downloaded [81%] (6s | 14Mb/s)
[INFO] 350 of 422 parts downloaded [83%] (6s | 14Mb/s)
[INFO] 360 of 422 parts downloaded [85%] (5s | 14Mb/s)
[INFO] 370 of 422 parts downloaded [88%] (4s | 14Mb/s)
[INFO] 380 of 422 parts downloaded [90%] (3s | 14Mb/s)
[INFO] 390 of 422 parts downloaded [92%] (2s | 14Mb/s)
[INFO] 400 of 422 parts downloaded [95%] (2s | 14Mb/s)
[INFO] 410 of 422 parts downloaded [97%] (1s | 15Mb/s)
[INFO] 420 of 422 parts downloaded [99%] (0s | 14Mb/s)
[INFO] 422 of 422 parts downloaded [100%] (0s | 14Mb/s)
[WARN] Subtitles downloading disabled for hardsubs streams.
[INFO] Subtitles downloading skipped!
[INFO][mkvmerge] Started merging

> "mkvmerge" -o "/home/shinji/multi-downloader-nx/videos/8 Man After (Sub)/S01E04 - Sachiko`s Decision [480p].mkv" --no-date --disable-track-statistics-tags --engage no_variable_data --video-tracks 0 --audio-tracks 1 --track-name 0:"Japanese [Uncut]" --language 1:jpn "/home/shinji/multi-downloader-nx/videos/8 Man After (Sub)/S01E04 - Sachiko`s Decision [480p].Japanese.ts" --no-subtitles --no-attachments
/bin/sh: 1: s: not found
mkvmerge v45.0.0 ('Heaven in Pennies') 64-bit
Error: No source files were given.
{
  status: 2,
  signal: null,
  output: [ null, null, null ],
  pid: 1259298,
  stdout: null,
  stderr: null,
  code: 2
}
[ERROR][mkvmerge] Merging failed with exit code 2
```

**Comment by izu-co** [comment]
Fixed in 936f86b1f0954148ab3d348d7e08531ad69e073c
